File: concordar/importers.py
# -*- coding: utf-8 -*-

from PyQt4 import QtGui

def graphical_tokenize(text):
   """Tokenizes the text using the document cursor."""

   def generate_list(text):
      doc = QtGui.QTextDocument(text)
      cursor = QtGui.QTextCursor(doc)
      cursor.select(QtGui.QTextCursor.WordUnderCursor)
      index = 0
      yield (index, cursor.position(), cursor.selectedText())
      while cursor.movePosition(QtGui.QTextCursor.NextWord):
         cursor.select(QtGui.QTextCursor.WordUnderCursor)
         index += 1
         yield (index, cursor.position(), cursor.selectedText())
   
   return tuple(generate_list(text))

File: concordar/importers.cpp
#include <vector> 
#include <QString>
#include <QTextDocument>
#include <QTextCursor>
#include "importers.h"

namespace importers {

    void graphical_tokenize(const QString& text, std::vector<concordance::Token>& tokens) {
    QTextDocument doc(text);
    QTextCursor cursor = QTextCursor(&doc);
    
    int index = 0;
    cursor.select(QTextCursor::WordUnderCursor);

    concordance::Token token;
    token.index = index;
    token.word= cursor.selectedText();
    token.position = cursor.position();
    tokens.push_back(token);

    while(cursor.movePosition(QTextCursor::NextWord)) {
        concordance::Token token;
        ++index;
        cursor.select(QTextCursor::WordUnderCursor);
        token.index = index;
        token.word= cursor.selectedText();
        token.position = cursor.position();
        tokens.push_back(token);
      
    }
}

}
